### PR TITLE
Clustering with bootstrap node

### PR DESCRIPTION
**Pull request by tgross**
This PR is the initial implementation.

Clustering using a bootstap node for discovery etcd requires some external service to bootstrap the clustering. When we have SRV support in Triton CNS we'll use that, but for now we'll stand up a single-node cluster and bootstrap the cluster using that. After the cluster is scaled-up we can remove the bootstrap node.

~~This implementation uses a shell script rather than ContainerPilot to shim the application because we don't have a way to inject the container IP address into the environment that we fork. This will be covered in https://github.com/joyent/containerpilot/issues/171 and then we'll eliminate this shell script.~~  Done.

~~TODO for this PR documentation in the README~~ Done.

cc @misterbisson

### CODE REVIEW

**Comment by tgross** [comment]
@misterbisson this should be ready to look at.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,55 @@
...
+
+This repo is a demonstration of [etcd](https://coreos.com/etcd/docs/latest/) designed for self-operation according to the [Autopilot pattern](http://autopilotpattern.io/).
+
+An etcd cluster needs an external source of data for all the nodes to find each other initially. This can be a bootstrap service (another etcd cluster) or an SRV record. Triton CNS does not yet support SRV records, so until it does we're standing up a temporary single-node cluster to bootstrap the cluster. After the cluster is scaled-up we can remove the bootstrap node.
+
+
+### Getting started
+
+1. [Get a Joyent account](https://my.joyent.com/landing/signup/) and [add your SSH key](https://docs.joyent.com/public-cloud/getting-started).
+1. Install the [Docker Toolbox](https://docs.docker.com/installation/mac/) (including `docker` and `docker-compose`) on your laptop or other environment, as well as the [Joyent Triton CLI](https://www.joyent.com/blog/introducing-the-triton-command-line-tool) (`triton` replaces our old `sdc-*` CLI tools)
```
Comment (misterbisson) [inline README.md]: Low priority: https://docs.joyent.com/public-cloud/api-access/cloudapi is a better choice for the Triton CLI install instructions

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+bootstrap:
+    image: autopilotpattern/etcd
+    mem_limit: 128m
+    ports:
+      - 4001:4001
+      - 2379:2379
+      - 2380:2380
+    hostname: bootstrap
+    labels:
+      - triton.cns.services=etcd
```
Comment (misterbisson) [inline docker-compose.yml]: I'm still reading through, but marking this to come back to: should this advertise the `etcd` service, or...?

Revision: does this instance join the cluster as a peer, or just bootstrap it? From the example in chat, it looks like it _doesn't_ join the cluster:

```
$ docker exec -it e_etcd_1 /usr/local/bin/etcdctl cluster-health
member 4622a9503b7416b is healthy: got healthy result from http://172.17.0.4:2379
member 8c39cb514af5f356 is healthy: got healthy result from http://172.17.0.5:2379
member 9a080bf931bf1e98 is healthy: got healthy result from http://172.17.0.3:2379
cluster is healthy
```

I would expect a fourth instance/member there, given the defaults in start.sh to scale to 1 bootstrap + 3 etcd.

Comment (tgross) [inline docker-compose.yml]: Nope, good catch. This node should not have a CNS service label. It doesn't join the cluster.

File: README.md (hunk 1 above)
Comment (tgross) [inline README.md]: Fixed.

File: docker-compose.yml (hunk 2 above)
Comment (tgross) [inline docker-compose.yml]: Fixed.

**Comment by misterbisson** [comment]
🏡 🚶

**Pull request merged**